src/lib/commands/run-on-cloud.ts

```diff
@@ -45,12 +47,31 @@ export async function runActorOrTaskOnCloud(apifyClient: ApifyClient, options: R
     let run: ActorRun;
 
     try {
-        if (localInput && type === 'Actor') {
+        if ((localInput || inputOverride) && type === 'Actor') {
+            let inputToUse: Record<string, unknown> | undefined;
+            let contentType: string;
+
+            if (inputOverride) {
+                inputToUse = inputOverride;
+                contentType = 'application/json';
+            } else if (localInput) {
+                const ext = mime.getExtension(localInput.contentType!);
+
+                if (ext === 'json') {
+                    inputToUse = JSON.parse(localInput.body.toString('utf8'));
+                    contentType = 'application/json';
+                } else {
+                    inputToUse = localInput.body as never;
+                    contentType = localInput.contentType!;
+                }
+            } else {
+                throw new Error('Unreachable');
+            }
```

Comment:
Couldn't this be a function that returns `inputToUse` and `contentType`?

Reply:
Probably could, but so far this is the only place where this is needed 🤷‍♂️. I'd probably refactor it later